src/lib/replication/log_store/home_raft_log_store.cpp

```diff
@@ -264,8 +264,8 @@ raft_buf_ptr_t HomeRaftLogStore::pack(ulong index, int32_t cnt) {
         [this, &out_buf, &remain_cnt]([[maybe_unused]] store_lsn_t cur, const log_buffer& entry) mutable -> bool {
             if (remain_cnt-- > 0) {
                 size_t avail_size = out_buf->size() - out_buf->pos();
-                if (avail_size < entry.size()) {
-                    avail_size += std::max(out_buf->size() * 2, (size_t)entry.size());
+                if (avail_size < entry.size() + sizeof(uint32_t)) {
```

Comment:
avail_size should be able to hold entry.size() and the length of this entry, which is a uint32